### PR TITLE
829272 - [others] UG documentation for caption and group summary rows.

### PR DESCRIPTION
**Pull request by sudarsanMS**
**Description**
we already have a UG documentation for table summary rows, we updated the caption and group summary rows in it.

### CODE REVIEW

File: MAUI/DataGrid/Summaries.md [hunk 1]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
 
 N> The Summary does not refresh with data. To update the summary for the newly added row, or for the modified summary column, set the [SfDataGrid.View.LiveDataUpdateMode](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Data.LiveDataUpdateMode.html) to `LiveDataUpdateMode.AllowDataShaping` or `LiveDataUpdateMode.AllowSummaryUpdate`.
 
+## Caption summaries
+
+The data grid provides built-in support for caption summaries. The caption summary value calculated based on the records in a group. The summary information will be displayed in the caption of group.
+
+### Formatting built-in caption summary
+
+By default, caption summary rows are displayed with the [SfDataGrid.GroupCaptionTextFormat]() property.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:33]: By default, the caption summary rows in the data grid are displayed using the [SfDataGrid.GroupCaptionTextFormat](https://chat.openai.com/) property.

The default group caption format is {ColumnName}: {Key} - {ItemsCount} Items.

ColumnName: This displays the name of the currently grouped column.
Key: This displays the key value of the group.
ItemsCount: This displays the number of items in a group.
DataGrid with formatting in caption summary row

You can customize the group caption text format by setting the SfDataGrid.GroupCaptionTextFormat property. The code example below illustrates how to customize the group caption text in the data grid:

File: MAUI/DataGrid/Summaries.md [hunk 2]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+{% tabs %}
+{% highlight xaml%}
+<sfGrid:SfDataGrid x:Name="dataGrid"
+                   AutoGenerateColumns="True"
+                   ColumnSizer="Star"
+                   GroupCaptionTextFormat="{}{ColumnName}: {Key}">
+{% endhighlight %}
+
+{% highlight c#%}
+//Customized group caption text
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:54]: Provide space after comment codes like,

// Customized group caption text

File: MAUI/DataGrid/Summaries.md [hunk 3]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+{% endtabs %}
+
+The following screenshot shows the outcome of the previous code:
+
+![DataGrid with formatting in caption summary row](Images\Caption-Summary\ShowSummaryInRow2.png)
+
+
+### Displaying summary for a row
+
+Display summary information in a row by setting the [DataGridSummaryRow.ShowSummaryInRow]() property to `true` and define summary columns. You have to define the [DataGridSummaryRow.Title]() based on the [DataGridSummaryColumn.Name]() property to format summary columns value in a row.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:66]: To display summary information in a row, you can set the [DataGridSummaryRow.ShowSummaryInRow](https://chat.openai.com/) property to true and define summary columns. Additionally, you need to define the [DataGridSummaryRow.Title](https://chat.openai.com/) based on the [DataGridSummaryColumn.Name](https://chat.openai.com/) property in order to format the values of the summary columns in the row.

File: MAUI/DataGrid/Summaries.md [hunk 4]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+{% endtabs %}
+
+The following screenshot shows the outcome for both values of `ShowSummaryInRow` to `true`:
+
+![DataGrid with customized caption summary row](Images\Caption-Summary\ShowSummaryInRow1.png)
+
+
+### Displaying summary for column
+
+Display summary information in the column by setting the `DataGridSummaryRow.ShowSummaryInRow` to `false` and define summary columns. `SfDataGrid.DataGridSummaryColumn` is the object of [DataGridSummaryRow.SummaryColumns]() collection that contains the following important properties:
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:116]: To display summary information in a column, you need to set the DataGridSummaryRow.ShowSummaryInRow property to false and define summary columns using the SfDataGrid.DataGridSummaryColumn object. The DataGridSummaryColumn object has several important properties:

[Name](https://chat.openai.com/): This property defines the name of the DataGridSummaryColumn, which is used to identify the column in the DataGridSummaryRow and provide a title for it.
[MappingName](https://chat.openai.com/): This property defines the corresponding column name used for the summary calculation.
[SummaryType](https://chat.openai.com/): This property defines the SummaryType enum property, which determines the type of aggregate function used for the summary calculation.
The DataGrid control provides several predefined aggregates, such as CountAggregate, Int32Aggregate, and DoubleAggregate. When the summary type is set as Custom, you can define a CustomAggregate class object to calculate custom summaries.

The [Format](https://chat.openai.com/) property defines a string property that formats the summary value and displays it. The Format property can have two parts separated by a colon (:). The first part denotes the aggregate function name, and the second part denotes the display format of the summary value.

Please refer to the "Formatting Summary" section for more information on how to format the summary, and the "Aggregate Types" section to learn about the different summary types.

In the following code snippet, a summary is defined for the Salary column:

File: MAUI/DataGrid/Summaries.md [hunk 5]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+  * Int32Aggregate
+  * DoubleAggregate
+
+[CustomAggregate]() defines the `CustomAggregate` class object when the summary type is set as `Custom` that calculates the custom summaries.
+
+The [Format]() defines the `string` property that formats the summary value and displays it. The `Format` property may contains two parts separated by a colon (:). First part denotes the aggregate function name, and second part denotes display format of the summary value.
+
+Refer to the [Formatting Summary](#_Formatting_Summary) section to know more about how to format summary, and [Aggregate Types](#_Aggregate_Types) section to know about different summary types.
+
+In the following code snippet, summary is defined for `Salary` column:
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:134]: To display summary information in a column, you need to set the DataGridSummaryRow.ShowSummaryInRow property to false and define summary columns using the SfDataGrid.DataGridSummaryColumn object. The DataGridSummaryColumn object has several important properties:

[Name](https://chat.openai.com/): This property defines the name of the DataGridSummaryColumn, which is used to identify the column in the DataGridSummaryRow and provide a title for it.
[MappingName](https://chat.openai.com/): This property defines the corresponding column name used for the summary calculation.
[SummaryType](https://chat.openai.com/): This property defines the SummaryType enum property, which determines the type of aggregate function used for the summary calculation.
The DataGrid control provides several predefined aggregates, such as CountAggregate, Int32Aggregate, and DoubleAggregate. When the summary type is set as Custom, you can define a CustomAggregate class object to calculate custom summaries.

The [Format](https://chat.openai.com/) property defines a string property that formats the summary value and displays it. The Format property can have two parts separated by a colon (:). The first part denotes the aggregate function name, and the second part denotes the display format of the summary value.

Please refer to the "Formatting Summary" section for more information on how to format the summary, and the "Aggregate Types" section to learn about the different summary types.

In the following code snippet, a summary is defined for the Salary column:

File: MAUI/DataGrid/Summaries.md [hunk 6]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+{% endhighlight c#%}
+{% endtabs %}
+
+![DataGrid with caption summary cells customized](Images\Caption-Summary\ShowSummaryInColumn.png)
+
+N> The `CaptionSummaryColumn` text will be aligned based on the `DataGridColumn.TextAlignment`.
+
+### Caption summary template
+
+The data grid hosts any view(s) inside a caption summary for the entire row or for individual columns by loading a template.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:170]: The data grid allows you to host any view(s) inside a caption summary for the entire row or for individual columns by loading a template

File: MAUI/DataGrid/Summaries.md [hunk 7]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+
+### Caption summary template
+
+The data grid hosts any view(s) inside a caption summary for the entire row or for individual columns by loading a template.  
+
+### Displaying template for a row
+
+The template for a caption summary row can be set by using [SfDataGrid.CaptionSummaryTemplate]() to customize it based on requirement. 
+
+Refer the below code example in which a label is loaded in the caption summary template of caption summary row.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:176]: The template for a caption summary row can be customized by using the [SfDataGrid.CaptionSummaryTemplate](https://chat.openai.com/) property. This allows you to define a custom template according to your requirements.

Please refer to the code example below, where a label is loaded in the caption summary template of the caption summary row:

File: MAUI/DataGrid/Summaries.md [hunk 8]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+![DataGrid with template loaded for entire caption summary row](Images\Caption-Summary\TemplateRow.png)
+
+N> The `DataTemplateSelector` can also be directly assigned to the `CaptionSummaryTemplate`.
+
+
+### Displaying template for a column
+
+The template for a Caption summary column can be set by using `DataGridSummaryColumn.Template` to customize it based on requirement. 
+
+Refer the below code example in which a label is loaded in the template of caption summary column.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:262]: The template for a caption summary column can be customized by using the DataGridSummaryColumn.Template property. This allows you to define a custom template according to your requirements.

Here's an example code snippet that demonstrates how to load a label in the template of a caption summary column:

File: MAUI/DataGrid/Summaries.md [hunk 9]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+{% endhighlight %}
+{% endtabs %}
+
+![DataGrid with template loaded for individual caption summary column](Images\Caption-Summary\TemplateColumn.png)
+
+## Group summary
+
+Group summary values are calculated based on records in the group. The summary information will be displayed at the bottom of each group. You can view the group summary row by expanding the corresponding group header. The data grid adds any number of group summary row.
+
+Add group summary rows in the data grid by adding the [DataGridSummaryRow]() to [SfDataGrid.GroupSummaryRows]() collection.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:348]: Group summary values are calculated based on the records within each group. The summary information will be displayed at the bottom of each group. You can view the group summary row by expanding the corresponding group header. The data grid allows for the addition of any number of group summary rows.

To add group summary rows in the data grid, you can add the [DataGridSummaryRow](https://chat.openai.com/) objects to the [SfDataGrid.GroupSummaryRows](https://chat.openai.com/) collection.

File: MAUI/DataGrid/Summaries.md [hunk 10]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+
+### Displaying summary in the row
+
+The summary information can be displayed in the row by setting the [DataGridSummaryRow.ShowSummaryInRow]() to `true` and by defining summary columns. You have to define the [DataGridSummaryRow.Title]() based on the [DataGridSummaryRow.Name]() property to format summary columns value in a row.
+ 
+Refer to [Formatting Summary](#formatting-summary) section to know more about how to format summary.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:354]: Summary information can be displayed in the entire row by setting the [DataGridSummaryRow.ShowSummaryInRow](https://chat.openai.com/) property to true and defining summary columns. The [DataGridSummaryRow.Title](https://chat.openai.com/) property needs to be defined based on the [DataGridSummaryColumn.Name](https://chat.openai.com/) property in order to format the values of the summary columns in a row.

Please refer to the [Formatting Summary](https://chat.openai.com/#formatting-summary) section to learn more about how to format the summary

File: MAUI/DataGrid/Summaries.md [hunk 11]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+
+The summary information can be displayed in the column by setting the [DataGridSummaryRow.ShowSummaryInRow]() to `false` and by defining summary columns. To calculate summary based on the column, specify the following properties:
+
+1. [DataGridSummaryColumn.MappingName](): Provides MappingName of the column (Property name of data object) that you want to calculate summary.
+2. [DataGridSummaryColumn.SummaryType](): Provides different built-in summary calculation functions for various types.
+3. [DataGridSummaryColumn.Format](): Provides format string for the summary based on support function name in the specified SummaryType.
+
+Refer to [Formatting Summary](#formatting-summary) section to know more about how to format summary and [Aggregate Types](#aggregate-types) section to know about different Summary Types.
+
+In the following code snippet, summary is defined for `Salary` and `CustomerID` columns:
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:427]: The summary information can be displayed in a column by setting the [DataGridSummaryRow.ShowSummaryInRow](https://chat.openai.com/) property to false and defining summary columns. To calculate the summary based on a column, you need to specify the following properties:

[DataGridSummaryColumn.MappingName](https://chat.openai.com/): This property provides the MappingName of the column (the property name of the data object) for which you want to calculate the summary.
[DataGridSummaryColumn.SummaryType](https://chat.openai.com/): This property provides different built-in summary calculation functions for various types.
[DataGridSummaryColumn.Format](https://chat.openai.com/): This property allows you to specify a format string for the summary based on the supported function name in the specified SummaryType.
Please refer to the "Formatting Summary" section to learn more about how to format the summary, and the "Aggregate Types" section to understand the different Summary Types.

In the following code snippet, a summary is defined for the Salary and CustomerID columns:

File: MAUI/DataGrid/Summaries.md [hunk 12]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+            });
+{% endhighlight %}
+{% endtabs %}
+
+
+![DataGrid with customized summary for group summary cells](Images\Group-Summary\ShowSummaryInColumn.png)
+
+### Group summary template
+
+The data grid hosts any view(s) inside a group summary for the entire row or for individual columns by loading a template.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:472]: The data grid allows you to host any view(s) inside a group summary for the entire row or individual columns by loading a template.

File: MAUI/DataGrid/Summaries.md [hunk 13]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+
+### Group summary template
+
+The data grid hosts any view(s) inside a group summary for the entire row or for individual columns by loading a template.  
+
+### Displaying template for a row
+
+The template for a group summary row can be set by using `SfDataGrid.GroupSummaryTemplate` to customize it based on requirement. 
+
+Refer the below code example in which a label is loaded in the group summary template of group summary row.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:478]: The template for a group summary row can be customized by using the SfDataGrid.GroupSummaryTemplate property.

Refer to the code example below, which demonstrates how to load a label in the group summary template of a group summary row:

File: MAUI/DataGrid/Summaries.md [hunk 14]
```diff
@@ -11,17 +11,647 @@ documentation: UG
...
+
+![DataGrid with template loaded for entire group summary row](Images\Group-Summary\RowTemplate.png)
+
+N> The `DataTemplateSelector` can also be directly assigned to the `SfDataGrid.GroupSummaryTemplate`.
+
+### Displaying template for a column
+
+The template for a group summary column can be set by using `GridSummaryColumn.Template` to customize it based on requirement. 
+
+Refer the below code example in which a label is loaded in the template of group summary column.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Summaries.md:569]: The template for a group summary column can be customized by using the GridSummaryColumn.Template property. This allows you to define a custom template according to your requirements.

Please refer to the code example below, which demonstrates how to load a label in the template of a group summary column:

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]
@sudarsanMS - I have added my comments. Please check and confirm.

File: MAUI/DataGrid/Summaries.md (hunk 1 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 2 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 3 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 5 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 6 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 7 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 8 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 9 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 10 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 11 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 12 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 13 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

File: MAUI/DataGrid/Summaries.md (hunk 14 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/Summaries.md]: Modified.

**Review by ashok-kuvaraja — APPROVED** [review]
The changes are fine.

**Review by FarjanaParveen — APPROVED** [review]
This is fine

**Pull request merged**